caffe2/CMakeLists.txt

```diff
@@ -1529,7 +1529,9 @@ if(USE_XPU)
   target_link_libraries(
       torch_xpu PRIVATE ${Caffe2_XPU_DEPENDENCY_LIBS})
 
-  target_link_libraries(torch_xpu PUBLIC torch_cpu_library)
+  target_link_libraries(
+      torch_xpu
+      PRIVATE "-Wl,--no-as-needed,\"$<TARGET_FILE:torch_cpu>\" -Wl,--as-needed")
```

Comment:
How can the line of code ensure that the torch cpu library is ready before torch_xpu build?

Reply:
Thanks for the reminder. Explicit dependency and cmake compiler checker are done.